dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/event/RequestParams.java

```diff
@@ -35,42 +35,59 @@
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
+import org.hisp.dhis.category.CategoryCombo;
+import org.hisp.dhis.category.CategoryOption;
 import org.hisp.dhis.common.AssignedUserSelectionMode;
 import org.hisp.dhis.common.IdSchemes;
+import org.hisp.dhis.common.OpenApi;
 import org.hisp.dhis.common.OrganisationUnitSelectionMode;
+import org.hisp.dhis.common.UID;
 import org.hisp.dhis.event.EventStatus;
 import org.hisp.dhis.fieldfiltering.FieldFilterParser;
 import org.hisp.dhis.fieldfiltering.FieldPath;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.hisp.dhis.program.Enrollment;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramStage;
 import org.hisp.dhis.program.ProgramStatus;
+import org.hisp.dhis.user.User;
 import org.hisp.dhis.webapi.controller.event.webrequest.PagingAndSortingCriteriaAdapter;
+import org.hisp.dhis.webapi.controller.tracker.view.Event;
+import org.hisp.dhis.webapi.controller.tracker.view.TrackedEntity;
 
 /**
  * Represents query parameters sent to {@link EventsExportController}.
  *
  * @author Giuseppe Nespolino <g.nespolino@gmail.com>
  */
+@OpenApi.Property
 @Data
 @NoArgsConstructor
 class RequestParams extends PagingAndSortingCriteriaAdapter
 {
     static final String DEFAULT_FIELDS_PARAM = "*,!relationships";
 
+    @OpenApi.Property( { UID.class, Program.class } )
     private String program;
 
+    @OpenApi.Property( { UID.class, ProgramStage.class } )
     private String programStage;
 
     private ProgramStatus programStatus;
 
     private Boolean followUp;
 
+    @OpenApi.Property( { UID.class, TrackedEntity.class } )
```

Comment:
Another question is what type to use here:

For metadata its simple as we only have one so I am using `org.hisp.dhis.program.Program`.

For tracker data its not so clear as we have the `org.hisp.dhis.webapi.controller.tracker.view.TrackedEntity;` representing the JSON we accept and return and the internal `org.hisp.dhis.trackedentity.TrackedEntity` used to persist.

@jbee and @enricocolasante which would you choose? I think I should stick to the JSON related ones in `org.hisp.dhis.webapi.controller.tracker.view.`

Reply:
I would say the view one as it is the one that the client interacts with